lectures/5-commitments.tex

```diff
@@ -0,0 +1,159 @@
+\end{remark}
+
+Pedersen commitment scheme algorithm:
+\begin{enumerate}
+    \item Prover and Verifier agrees on $G$ and $U$ points
+    \item Prover selects a value $m$ to commit and a blinder factor $r$:
+            $m \leftarrow \mathbb{Z}$,
+            $r \xleftarrow{R} \mathbb{Z}$
+    \item Prover generates a commitment and sends it to the Verifier:
+            $c = mG + rU$
+\end{enumerate}
+
+During the verification stage, prover reveals $(m, r)$ to the verifier. 
+To check the commitment, verifier computes: $c_1 = mG + rU$.
+
+If $c_1 = c$, prover has revealed the correct pair $(m, r)$.
+
+\begin{remark}
+    In case the discrete logarithm of $U$ is leaked, the \textit{binding} property can be violated by the \textit{Prover}:
+
+    $c = mG + rU = m \cdot G + r*u \cdot G = (m + r*u) \cdot G$
+    
+    For example, $(m + u, r - 1)$ will have the same commitment value:
+
+    $(m+u + (r-1)*u) \cdot G = (m + u - u + r*u) \cdot G = (m + r*u) \cdot G$ 
+        
+\end{remark}
+
+\textbf{Commitment aggregation}
+
+Pedersen commitment have some advantages compared to hash-based commitments.
+Additively homomorphic property allows to accumulate multiple commitments into one.
+Consider two pairs: $(m_1, r_1), (m_2, r_2)$.
+
+\begin{center}
+    $C_1 = m_1G + r_1U$,
+
+    $C_2 = m_2G + r_2U$,
+
+    $C_a = C_1 + C_2 = (m_1 + m_2)G + (r_1 + r_2)U$
+\end{center}
+
+This works for any number of commitments, so we can encode as many points as we like in a single one.
+
+\vspace{0.5 cm}
+
+\subsubsection{Vector commitments}
+
+Vector commitment schemes allows to commit to a vector of values rather than a value and a blinding term.
+
+\vspace{0.5 cm}
+
+\textbf{Pedersen Vector Commitments}
+
+Suppose we have a set of random elliptic curve points $(G_1,...,G_n)$ of group $\mathbb{G}$ 
+(that we do not know the discrete logarithm of), a vector $[m_1, m_2 ... m_n]$ and a random value $r$. We can do the following:
+
+\begin{center}
+    $ [C] = m_1\cdot[G_1] + m_2 \cdot [G_2] ... + m_n \cdot [G_n] + r \cdot [Q]$ 
+\end{center}
+
+Since the \textit{Prover} does not know the discrete logarithm of the generators, they don`t know the discrete logarithm of [C]. Hence, this scheme is binding: they can only reveal $(v_1,...,v_n)$ to produce $[C]$ later, they cannot produce another vector.
+
+Prover can later open the commitment by revealing the vector $[m_1, m_2 ... m_n]$ and a blinding term $r$.
+
+\vspace{0.5 cm}
+
+\textbf{Merkle Tree based Vector Commitments}
+
+A naive approach for a vector commitment would be hash the whole vector. More sophisticated scheme uses divide-and-conquer approach by building a binary tree out of vector elements.
+
+\begin{center}
```

Comment:
Better to use `\begin{figure}` environment (one you are using here is for figures inside the `\begin{example}` block)